src/imagery/i.sam2/i.sam2.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+############################################################################
+#
+# MODULE:       i.sam2
+# AUTHOR:       Corey T. White, OpenPlains Inc.
+# PURPOSE:      Uses the SAMGeo model for segmentation in GRASS GIS.
+# COPYRIGHT:    (C) 2023-2024 Corey White
+#               This program is free software under the GNU General
+#               Public License (>=v2). Read the file COPYING that
+#               comes with GRASS for details.
+#
+#############################################################################
+
+# %module
+# % description: Integrates SAMGeo model with text prompt for segmentation in GRASS GIS.
+# % keyword: imagery
+# % keyword: segmentation
+# % keyword: object recognition
+# % keyword: deep learning
+# %end
+
+# %option G_OPT_I_GROUP
+# % key: group
+# % description: Name of input imagery group
+# % required: yes
+# %end
+
+# %option
+# % key: output
+# % type: string
+# % description: Name of output segmented raster map
+# % gisprompt: new,cell,raster
+# % required: yes
+# %end
+
+# %option
+# % key: model_path
+# % type: string
+# % description: Path to the SAMGeo model file (optional if using default model)
+# % required: no
+# %end
+
+# %option
+# % key: text_prompt
+# % type: string
+# % description: Optional text prompt to guide segmentation
+# % required: yes
+# %end
+
+# %option
+# % key: text_threshold
+# % type: double
+# % answer: 0.24
+# % description: Text threshold for text segmentation
+# % required: no
+# % multiple: no
+# %end
+
+# %option
+# % key: box_threshold
+# % type: double
+# % answer: 0.24
+# % description: Box threshold for text segmentation
+# % required: no
+# % multiple: no
+# %end
+
+
+import os
+import sys
+import grass.script as gs
+import torch
+import numpy as np
+from PIL import Image
+from grass.script import array as garray
+
+
+def main():
+    from samgeo import SamGeo
+    from samgeo.text_sam import LangSAM
+
+    group = options["group"]
+
+    output_raster = options["output"]
+    model_path = options.get("model_path")
+    text_prompt = options.get("text_prompt")
+    text_threshold = float(options.get("text_threshold"))
+    box_threshold = float(options.get("box_threshold"))
+
+    # Set default model path if not provided
+    default_model_path = os.path.join(
+        gs.gisenv()["GISDBASE"], "samgeo_default_model.pth"
+    )
+    if not model_path:
+        model_path = default_model_path
+
+    # Set up paths to access the raster files
+    tmp_dir = gs.tempdir()
+    temp_input_path = os.path.join(tmp_dir, "input.tif")
+
+    temp_output_path = tmp_dir
+    guide_input_path = None
+
+    rasters = gs.read_command("i.group", group=group, flags="lg").strip().split("\n")
```

Comment:
In windows at least, the elements in the resulting list 'rasters' have a trailing '\r', for example: ['lsat5_1987_10@landsat\r', 'lsat5_1987_20@landsat\r', 'lsat5_1987_30@landsat']

Reply:
@cwhite911 The issue mentioned above can be resolved, I think, by replacing line 105 by:

```
rasters = gs.read_command("i.group", group=group, flags="lg")
rasters = [raster.strip() for raster in rasters.splitlines()]
```

Reply:
This is updated now if you want to test it out again.